plugins/modules/cluster_role.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Paul Arthur <paul.arthur@flowerysong.com>
+# Copyright: (c) 2019, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'XLAB Steampunk'}
+
+DOCUMENTATION = '''
+module: cluster_role
+author:
+ - Paul Arthur (@flowerysong)
+ - Manca Bizjak (@mancabizjak)
+ - Aljaz Kosir (@aljazkosir)
+ - Tadej Borovsak (@tadeboro)
+short_description: Manages Sensu cluster roles
+description:
+  - For more information, refer to the Sensu documentation at
+    U(https://docs.sensu.io/sensu-go/latest/reference/rbac/)
+extends_documentation_fragment:
+  - sensu.sensu_go.auth
+  - sensu.sensu_go.name
+  - sensu.sensu_go.state
+notes:
+  - Parameter C(auth.namespace) is ignored in this module.
+options:
+  rules:
```

Comment:
Does sensu API accepts `rules: []`? And does ansible accept that thing as valid? This is getting a bit tricky ;)

Reply:
Not valid on sensu API, but Ansible treats it as valid, even though `argument_spec` says that `verbs` and `resources` are required. If we set `required=True` on `rules` problems goes away... but `absent` state requires `rules` too then.
I think creating a custom validator function would be the best thing to do right here?

Also while I was looking into this I found out that a valid `verb` is also `['*']` which allows all permissions.

Reply:
I also asked sensu devs about the semantics of the `*` value. Once we hear what they have to say about this, we will act appropriately, but for now, I would say that we just add it as a valid option and be done with it.

Reply:
As for the roles being required, I vote for adding a custom validation into the module. This way we can at least produce a sensible error message while sensu API will probably just return a numeric error.

Reply:
Yeah, I'm greeted with classic 400 Bad request :grin: